build.rs

```diff
@@ -66,10 +69,17 @@ fn mm_version() -> String {
     let mm_version_p = root().join("MM_VERSION");
     let v_file = String::from_utf8(slurp(&mm_version_p)).unwrap();
     let v_file = v_file.trim().to_string();
-    if version[..] != v_file[..] {
-        // Create or update the MM_VERSION file in order to appease the Cargo dependency management.
-        let mut mm_version_f = fs::File::create(&mm_version_p).unwrap();
-        mm_version_f.write_all(version.as_bytes()).unwrap();
+    // if there is no MM_VERSION file there is no need to create it
+    if !v_file.is_empty() {
+        if !v_file.contains(&version) {
+            // If the file doesn't contain the latest commit hash then this is a local build and the file should be updated
+            let mut mm_version_f = fs::File::create(&mm_version_p).unwrap();
```

Comment:
It doesn't match the requirement from Discord DM: `5. If MM_VERSION exists, check if it contains the commit hash. If it doesn't, append commit hash to env only. Do not change the file.` Doing it this way will allow us to put something like `ARTEM_DEV_BUILD` on our workstations, and the commit hash will be appended automatically. It will help to distinguish logs from local builds. Also, some of the notary operators may like to set the personalized version without bothering about updating the file every time.

Reply:
So the file shouldn't change in all the cases? 
1 - If the file isn't there it shouldn't be created.
2 - If MM_VERSION exists, and it doesn't contain the commit hash. The file shouldn't change as per the above comment.
3 - If MM_VERSION exists and it contains the commit hash, it will have been probably created by the CI e.g. `2.1.3665_mm2.1_419f36ef5_Linux_Release` since the CI creates the file before the build script is run. The file shouldn't be changed in this case either.

Is this correct? or am I missing something?

Reply:
> Is this correct? or am I missing something?

Yes, it's correct :slightly_smiling_face: